python-frontend/src/main/java/org/sonar/python/semantic/ProjectLevelSymbolTable.java

```diff
@@ -138,24 +137,13 @@ public Symbol getSymbol(@Nullable String fullyQualifiedName) {
   }
 
   public Symbol getSymbol(@Nullable String fullyQualifiedName, @Nullable String localSymbolName) {
+    return getSymbol(fullyQualifiedName, localSymbolName, new HashMap<>());
+  }
+
+  public Symbol getSymbol(@Nullable String fullyQualifiedName, @Nullable String localSymbolName, Map<String, Symbol> createdSymbols) {
     if (fullyQualifiedName == null) return null;
-    if (queriedSymbolNames.contains(fullyQualifiedName)) {
-      // cyclic dependencies
-      queriedSymbolNames = new HashSet<>();
-      String[] fqnSplitByDot = fullyQualifiedName.split("\\.");
-      localSymbolName =  localSymbolName != null ? localSymbolName : fqnSplitByDot[fqnSplitByDot.length - 1];
-      return new SymbolImpl(localSymbolName, fullyQualifiedName);
-    }
     Descriptor descriptor = globalDescriptorsByFQN().get(fullyQualifiedName);
-    if (descriptor == null) {
```

Comment:
that's great that `queriedSymbolsNames` got removed!

Reply:
Indeed! As you suspected, it really wasn't a good solution in the end.